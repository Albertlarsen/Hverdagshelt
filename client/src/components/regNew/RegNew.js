import {Grid, Row, Col, PageHeader, Button} from 'react-bootstrap';
import {Component} from 'react';
import * as React from 'react';
import {history} from "../../index";


export class RegNew extends Component<Props, State>{
    render(){
        return(
        <Grid>
            <Col md={3}>
            </Col>
            <Col md={6}>
                <Col md={12}>
                    <PageHeader>
                    Bruker ble registrert!

                        <small>Email med passord er sendt</small>
                    </PageHeader>
                </Col>
                <Col md={12}>

                </Col>
                <Col md={12}>
                    <Button type="button" onClick={this.goToRegisterAdmin}>Lag ny admin</Button>
                    <Button type="button" onClick={this.goToRegisterEmployee}>Lag ny kommuneansatt</Button>
                    <Button type="button" onClick={this.goToMinSide}>GÃ¥ til min side</Button>
                </Col>
            </Col>
            <Col md={3}/>
        </Grid>
        )
    }
    goToRegisterAdmin = () => {
        history.push('/registrer/admin');
    };
    goToRegisterEmployee = () =>{
        history.push('/registrer/kommuneansatt')
    };
    goToMinSide = () =>{
        history.push('/admin')
    };
}