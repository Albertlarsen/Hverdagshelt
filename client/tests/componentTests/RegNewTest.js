// @flow

import {shallow} from "enzyme/build";
import * as React from "react";
import {RegNew} from "../../src/components/regNew/RegNew";

describe('Test for RegNew component', () => {
    const wrapper = shallow(<RegNew />);


    it('renders correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

});