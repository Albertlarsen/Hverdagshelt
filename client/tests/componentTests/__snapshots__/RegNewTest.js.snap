// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for RegNew component renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RegNew />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "bsClass": "container",
      "children": Array [
        <Col
          bsClass="col"
          componentClass="div"
          md={3}
        />,
        <Col
          bsClass="col"
          componentClass="div"
          md={6}
        >
          <Col
            bsClass="col"
            componentClass="div"
            md={12}
          >
            <PageHeader
              bsClass="page-header"
            >
              Bruker ble registrert!
              <small>
                Email med passord er sendt
              </small>
            </PageHeader>
          </Col>
          <Col
            bsClass="col"
            componentClass="div"
            md={12}
          />
          <Col
            bsClass="col"
            componentClass="div"
            md={12}
          >
            <Button
              active={false}
              block={false}
              bsClass="btn"
              bsStyle="default"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              Lag ny admin
            </Button>
            <Button
              active={false}
              block={false}
              bsClass="btn"
              bsStyle="default"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              Lag ny kommuneansatt
            </Button>
            <Button
              active={false}
              block={false}
              bsClass="btn"
              bsStyle="default"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              Gå til min side
            </Button>
          </Col>
        </Col>,
        <Col
          bsClass="col"
          componentClass="div"
          md={3}
        />,
      ],
      "componentClass": "div",
      "fluid": false,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bsClass": "col",
          "componentClass": "div",
          "md": 3,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bsClass": "col",
          "children": Array [
            <Col
              bsClass="col"
              componentClass="div"
              md={12}
            >
              <PageHeader
                bsClass="page-header"
              >
                Bruker ble registrert!
                <small>
                  Email med passord er sendt
                </small>
              </PageHeader>
            </Col>,
            <Col
              bsClass="col"
              componentClass="div"
              md={12}
            />,
            <Col
              bsClass="col"
              componentClass="div"
              md={12}
            >
              <Button
                active={false}
                block={false}
                bsClass="btn"
                bsStyle="default"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                Lag ny admin
              </Button>
              <Button
                active={false}
                block={false}
                bsClass="btn"
                bsStyle="default"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                Lag ny kommuneansatt
              </Button>
              <Button
                active={false}
                block={false}
                bsClass="btn"
                bsStyle="default"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                Gå til min side
              </Button>
            </Col>,
          ],
          "componentClass": "div",
          "md": 6,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "col",
              "children": <PageHeader
                bsClass="page-header"
              >
                Bruker ble registrert!
                <small>
                  Email med passord er sendt
                </small>
              </PageHeader>,
              "componentClass": "div",
              "md": 12,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "page-header",
                "children": Array [
                  "Bruker ble registrert!",
                  <small>
                    Email med passord er sendt
                  </small>,
                ],
              },
              "ref": null,
              "rendered": Array [
                "Bruker ble registrert!",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Email med passord er sendt",
                  },
                  "ref": null,
                  "rendered": "Email med passord er sendt",
                  "type": "small",
                },
              ],
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "col",
              "componentClass": "div",
              "md": 12,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "col",
              "children": Array [
                <Button
                  active={false}
                  block={false}
                  bsClass="btn"
                  bsStyle="default"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  Lag ny admin
                </Button>,
                <Button
                  active={false}
                  block={false}
                  bsClass="btn"
                  bsStyle="default"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  Lag ny kommuneansatt
                </Button>,
                <Button
                  active={false}
                  block={false}
                  bsClass="btn"
                  bsStyle="default"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  Gå til min side
                </Button>,
              ],
              "componentClass": "div",
              "md": 12,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "active": false,
                  "block": false,
                  "bsClass": "btn",
                  "bsStyle": "default",
                  "children": "Lag ny admin",
                  "disabled": false,
                  "onClick": [Function],
                  "type": "button",
                },
                "ref": null,
                "rendered": "Lag ny admin",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "active": false,
                  "block": false,
                  "bsClass": "btn",
                  "bsStyle": "default",
                  "children": "Lag ny kommuneansatt",
                  "disabled": false,
                  "onClick": [Function],
                  "type": "button",
                },
                "ref": null,
                "rendered": "Lag ny kommuneansatt",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "active": false,
                  "block": false,
                  "bsClass": "btn",
                  "bsStyle": "default",
                  "children": "Gå til min side",
                  "disabled": false,
                  "onClick": [Function],
                  "type": "button",
                },
                "ref": null,
                "rendered": "Gå til min side",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bsClass": "col",
          "componentClass": "div",
          "md": 3,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bsClass": "container",
        "children": Array [
          <Col
            bsClass="col"
            componentClass="div"
            md={3}
          />,
          <Col
            bsClass="col"
            componentClass="div"
            md={6}
          >
            <Col
              bsClass="col"
              componentClass="div"
              md={12}
            >
              <PageHeader
                bsClass="page-header"
              >
                Bruker ble registrert!
                <small>
                  Email med passord er sendt
                </small>
              </PageHeader>
            </Col>
            <Col
              bsClass="col"
              componentClass="div"
              md={12}
            />
            <Col
              bsClass="col"
              componentClass="div"
              md={12}
            >
              <Button
                active={false}
                block={false}
                bsClass="btn"
                bsStyle="default"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                Lag ny admin
              </Button>
              <Button
                active={false}
                block={false}
                bsClass="btn"
                bsStyle="default"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                Lag ny kommuneansatt
              </Button>
              <Button
                active={false}
                block={false}
                bsClass="btn"
                bsStyle="default"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                Gå til min side
              </Button>
            </Col>
          </Col>,
          <Col
            bsClass="col"
            componentClass="div"
            md={3}
          />,
        ],
        "componentClass": "div",
        "fluid": false,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "col",
            "componentClass": "div",
            "md": 3,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "col",
            "children": Array [
              <Col
                bsClass="col"
                componentClass="div"
                md={12}
              >
                <PageHeader
                  bsClass="page-header"
                >
                  Bruker ble registrert!
                  <small>
                    Email med passord er sendt
                  </small>
                </PageHeader>
              </Col>,
              <Col
                bsClass="col"
                componentClass="div"
                md={12}
              />,
              <Col
                bsClass="col"
                componentClass="div"
                md={12}
              >
                <Button
                  active={false}
                  block={false}
                  bsClass="btn"
                  bsStyle="default"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  Lag ny admin
                </Button>
                <Button
                  active={false}
                  block={false}
                  bsClass="btn"
                  bsStyle="default"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  Lag ny kommuneansatt
                </Button>
                <Button
                  active={false}
                  block={false}
                  bsClass="btn"
                  bsStyle="default"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  Gå til min side
                </Button>
              </Col>,
            ],
            "componentClass": "div",
            "md": 6,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "col",
                "children": <PageHeader
                  bsClass="page-header"
                >
                  Bruker ble registrert!
                  <small>
                    Email med passord er sendt
                  </small>
                </PageHeader>,
                "componentClass": "div",
                "md": 12,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "bsClass": "page-header",
                  "children": Array [
                    "Bruker ble registrert!",
                    <small>
                      Email med passord er sendt
                    </small>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Bruker ble registrert!",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Email med passord er sendt",
                    },
                    "ref": null,
                    "rendered": "Email med passord er sendt",
                    "type": "small",
                  },
                ],
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "col",
                "componentClass": "div",
                "md": 12,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "col",
                "children": Array [
                  <Button
                    active={false}
                    block={false}
                    bsClass="btn"
                    bsStyle="default"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                  >
                    Lag ny admin
                  </Button>,
                  <Button
                    active={false}
                    block={false}
                    bsClass="btn"
                    bsStyle="default"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                  >
                    Lag ny kommuneansatt
                  </Button>,
                  <Button
                    active={false}
                    block={false}
                    bsClass="btn"
                    bsStyle="default"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                  >
                    Gå til min side
                  </Button>,
                ],
                "componentClass": "div",
                "md": 12,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "active": false,
                    "block": false,
                    "bsClass": "btn",
                    "bsStyle": "default",
                    "children": "Lag ny admin",
                    "disabled": false,
                    "onClick": [Function],
                    "type": "button",
                  },
                  "ref": null,
                  "rendered": "Lag ny admin",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "active": false,
                    "block": false,
                    "bsClass": "btn",
                    "bsStyle": "default",
                    "children": "Lag ny kommuneansatt",
                    "disabled": false,
                    "onClick": [Function],
                    "type": "button",
                  },
                  "ref": null,
                  "rendered": "Lag ny kommuneansatt",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "active": false,
                    "block": false,
                    "bsClass": "btn",
                    "bsStyle": "default",
                    "children": "Gå til min side",
                    "disabled": false,
                    "onClick": [Function],
                    "type": "button",
                  },
                  "ref": null,
                  "rendered": "Gå til min side",
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "col",
            "componentClass": "div",
            "md": 3,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
